### DISCUSSION TITLE
Borg Backup Config went missing

### DISCUSSION THREAD

**Discussion by theOne-Q** [nextcloud/all-in-one#2837] [Questions] [answered]
c

### Steps to reproduce
1. use NC 26
2. click create Backup in AIO 
3. observe Error

### Expected behavior 
start creating a Backup
### Actual behavior 
abort  Backup

### Host OS 
Fedora Server with Docker AIO succesfully installed

#### Nextcloud AIO version 
beta 6.2.0
#### Current channel 
beta
#### Other valuable info 
"
Performing backup...
Did not find borg.config file in the mastercontainer volume.
Cannot create a backup as this is wrong.
"

**Comment by szaimen** [comment]
Hi, I am not completely sure how the file got missing on your system as I cannot reproduce this. Did you maybe delete the file at some point manually? I also did another look at the logic and could not find a way that could lead to this.

Anyway, you should be able to fix this by copying the `config` file from the borg backup directory into `/var/lib/docker/volumes/nextcloud_aio_mastercontainer/_data/data/borg.config`

**Comment by theOne-Q** [comment]
Thank you, but i cant find that config file...where am i supposed to find it ? i cant find it in the Github files. 
Also i never manually deleted it....I truly never touched the containers or volumes

**Reply by szaimen** [reply]
All right, I believe you. However I cannot reproduce this.

I'll try to help you to fix this going forward. Can you post the output of `sudo find / -type f -name config` here?

**Reply by theOne-Q** [reply]
should I execute this seach command in root dir ?

**Reply by theOne-Q** [reply]
btw thanks for your help....like really

**Reply by szaimen** [reply]
You need to log in via shell on the server and then  run the command.

**Reply by theOne-Q** [reply]
alright... the output loads quite long. it seems like its still searching but here is what i have now:

/sys/devices/pci0000:00/0000:00:1f.2/config
/sys/devices/pci0000:00/0000:00:1f.0/config
/sys/devices/pci0000:00/0000:00:16.0/config
/sys/devices/pci0000:00/0000:00:1b.0/config
/sys/devices/pci0000:00/0000:00:19.0/config
/sys/devices/pci0000:00/0000:00:1f.3/config
/sys/devices/pci0000:00/0000:00:00.0/config
/sys/devices/pci0000:00/0000:00:03.0/config
/sys/devices/pci0000:00/0000:00:1a.0/config
/sys/devices/pci0000:00/0000:00:16.3/config
/sys/devices/pci0000:00/0000:00:1d.0/config
/sys/devices/pci0000:00/0000:00:1f.6/config
/sys/devices/pci0000:00/0000:00:02.0/config
/sys/devices/pci0000:00/0000:00:14.0/config
/etc/crypto-policies/config
/etc/selinux/config
/var/lib/docker/volumes/nextcloud_aio_nextcloud/_data/custom_apps/deck/.tx/config
/var/lib/docker/volumes/nextcloud_aio_nextcloud/_data/custom_apps/richdocuments/.tx/config
/var/lib/docker/volumes/nextcloud_aio_nextcloud/_data/custom_apps/groupfolders/.tx/config
/var/lib/docker/volumes/nextcloud_aio_nextcloud/_data/custom_apps/notes/.tx/config
/var/lib/docker/overlay2/42aad9bcef7a68608006f586b61abc230bb4943d3a4fd595b737b20a7eec69db/merged/var/www/docker-aio/.git/config
/var/lib/docker/overlay2/b319c11c3563e686bcc225482206773aeb229df6c0a60ad87c60fb37dc180ace/diff/tmp/nextcloud-aio/.git/config
/var/lib/docker/overlay2/31acb4fe0c156cc37f5ba2ea02acfb2f268ac0d0770bbf3c65db694c024896b1/diff/var/www/docker-aio/.git/config
/var/lib/docker/overlay2/cee2b6516e8994c82249c489ba405c9395ba2227c9e678a0ca6c0100b80f89e1/diff/tmp/nextcloud-aio/.git/config
/var/lib/docker/overlay2/68ac3ee280cf80eb09e0666f9b32501abb6cbf117bcf3ccb1fb1d46f60a06a61/merged/tmp/nextcloud-aio/.git/config
/var/lib/docker/overlay2/cb79ce846eeb58a8621ca28b46553f4a29a342ee91ea45bf8f0395efdba03105/diff/tmp/nextcloud-aio/.git/config
/usr/lib/modules/6.2.15-300.fc38.x86_64/config
/usr/lib/modules/6.3.6-200.fc38.x86_64/config
/usr/lib/modules/6.3.8-200.fc38.x86_64/config

**Reply by theOne-Q** [reply]
would it not be possible to copy the config from the github to my server...it should be the same every wheree right ?

**Reply by theOne-Q** [reply]
uhhh

/mnt/speicher_backup/newBackup/borg/config

this is what took so long.....tada there it is

**Reply by szaimen** [reply]
> /mnt/speicher_backup/newBackup/borg/config

yes, please copy that one into the mastercontainer volume via:
```
cp /mnt/speicher_backup/newBackup/borg/config /var/lib/docker/volumes/nextcloud_aio_mastercontainer/_data/data/borg.config
```

Afterwarrds backup creation should work again.

**Reply by theOne-Q** [reply]
alright thank you sir.
you saved my evening :)

**Reply by theOne-Q** [reply]
I`ll report if  it worked

**Reply by theOne-Q** [reply]
it now gets past the error of not finding the file but now it has issues with the keys :


Performing backup...
Starting the backup...
passphrase supplied in BORG_PASSPHRASE, by BORG_PASSCOMMAND or via BORG_PASSPHRASE_FD is incorrect.
Deleting the failed backup archive...
passphrase supplied in BORG_PASSPHRASE, by BORG_PASSCOMMAND or via BORG_PASSPHRASE_FD is incorrect.
Backup failed!
You might want to check the backup integrity via the AIO interface.



I am now running the Backup integrity check from AIO


Edit: this may take a while

**Reply by szaimen** [reply, marked as answer]
Interesting! I think I know what happened. You apparently in the past already created a backup with an old aio instance in /mnt/speicher_backup/newBackup. Then you created a new AIO instance and chose /mnt/speicher_backup/newBackup as backup target as well. Now the script runs into this problem because this is not expected.

**Reply by szaimen** [reply]
To proof that, can you post the output of the following command?
```
grep 'id =' /var/lib/docker/volumes/nextcloud_aio_mastercontainer/_data/data/borg.config
grep 'id =' /mnt/speicher_backup/newBackup/borg/config
```

**Reply by theOne-Q** [reply]
id = 93fe17392b6fe0b5f838dcf7e59b52a169f849f808282ae0b221ad3193f074a4
id = 93fe17392b6fe0b5f838dcf7e59b52a169f849f808282ae0b221ad3193f074a4

**Reply by szaimen** [reply]
Ah sorry, that of course would be the same as we just copied it 🙈

**Reply by theOne-Q** [reply]
but your right regardles...I did exactly what you described

**Reply by theOne-Q** [reply]
the qestion now is ....how to i make it ignore the keys or passphraes, or change them ... or do i need to delete all my old backups ?

**Reply by szaimen** [reply]
> .I did exactly what you described

All right, good to know then :)

You can do the following to fix the problem:
```
rm /var/lib/docker/volumes/nextcloud_aio_mastercontainer/_data/data/borg.config
mv /mnt/speicher_backup/newBackup/borg /mnt/speicher_backup/newBackup/borg-old
```

Afterwards the backup creation should work again.

**Reply by theOne-Q** [reply]
uhhhh

mv: cannot move '/mnt/speicher_backup/newBackup/borg' to 'mnt/speicher_backup/newBackup/borg-old': No such file or directory

**Reply by szaimen** [reply]
yeah sorry I made a typo. please try again with the adjusted command

**Reply by theOne-Q** [reply]
it did not throw an error :)

**Reply by theOne-Q** [reply]
uhm:



initializing repository...
using builtin fallback logging configuration
33 self tests completed in 0.21 seconds
Initializing repository at "/mnt/borgbackup/borg"
Key in "<Repository /mnt/borgbackup/borg>" created.
Keep this key safe. Your data will be inaccessible without it.
check_free_space: few segments, not requiring a full free segment
check_free_space: calculated working space for compact as 0 bytes
check_free_space: required bytes 169138, free bytes 3466581180416
security: previous location file /root/.config/borg/security/b47f56996c8b03b660f61e34042ae46b8952831f404a7b8c052ea406916d1b87/location not found
security: manifest timestamp file /root/.config/borg/security/b47f56996c8b03b660f61e34042ae46b8952831f404a7b8c052ea406916d1b87/manifest-timestamp not found
security: determined newest manifest timestamp as 
security: remembering previously unknown repository
security: saving state for b47f56996c8b03b660f61e34042ae46b8952831f404a7b8c052ea406916d1b87 to /root/.config/borg/security/b47f56996c8b03b660f61e34042ae46b8952831f404a7b8c052ea406916d1b87
security: current location   /mnt/borgbackup/borg
security: key type           5
security: manifest timestamp 2023-06-25T19:58:01.089601
security: repository checks ok, allowing access
Synchronizing chunks cache...
Archives: 0, w/ cached Idx: 0, w/ outdated Idx: 0, w/o cached Idx: 0.
Verified integrity of /mnt/borgbackup/borg/index.1
Done.
security: saving state for b47f56996c8b03b660f61e34042ae46b8952831f404a7b8c052ea406916d1b87 to /root/.config/borg/security/b47f56996c8b03b660f61e34042ae46b8952831f404a7b8c052ea406916d1b87
security: current location   /mnt/borgbackup/borg
security: key type           5
security: manifest timestamp 2023-06-25T19:58:01.089601

By default repositories initialized with this version will produce security
errors if written to with an older version (up to and including Borg 1.0.8).

If you want to use these older versions, you can disable the check by running:
borg upgrade --disable-tam /mnt/borgbackup/borg

See https://borgbackup.readthedocs.io/en/stable/changes.html#pre-1-0-9-manifest-spoofing-vulnerability for details about the security implications.

IMPORTANT: you will need both KEY AND PASSPHRASE to access this repo!
If you used a repokey mode, the key is stored in the repo, but you should back it up separately.
Use "borg key export" to export the key, optionally in printable format.
Write down the passphrase. Store both at safe place(s).




it warns me about it but it doesnt give me the keys or the phrases....not in the log or the AIO 

what now XD ?


Do i have to extract them from "<Repository /mnt/borgbackup/borg>" ??


Well it now creates a backup from my about 400 Gigs of data....so that worked :)


I guess AIO will show me the Key and Phrase when its done ???

**Reply by szaimen** [reply]
> it warns me about it but it doesnt give me the keys or the phrases....not in the log or the AIO

The passphrase is shown to you in the AIO interface. They key is in the config file afaik. Anyway, for AIO to restore from borg backup, you usually only need the passphrase that is shown to you in the AIO interface (and the borg archive obviously).

**Reply by theOne-Q** [reply]
SIR.....THANK YOU .

send me your Paypal i will donate you 5 bucks


you saved my day....seriously :)

**Reply by szaimen** [reply]
My paypal mail address is the one that is shown on my github profile :)

**Reply by theOne-Q** [reply]
SENT !

**Reply by szaimen** [reply]
Thank you! :)

**Reply by tailsec** [reply]
For anyone running via Docker Desktop on Windows like me, the command to run to copy the file is a little different based on the inability to copy directly to a volume - you need to copy to the master container's mount:

```
docker cp "C:\backup\borg\config" nextcloud-aio-mastercontainer:/mnt/docker-aio-config/data/borg.config
```

Where `"C:\backup\borg\config"` is the location of the backed up config file

**Comment by Gabemedina23** [comment]
I am an idiot... I had external storage and mistakenly set the backup to local storage that wasn't large enough to handle the entire nextcloud instance so when I went to change the backup location it said it couldn't write to the config due to lack of storage. In my mind that was as simple as deleting the '/mnt/backup/data folder' when I went to change it after this point it saved the changes to the config... well some time later i forget the steps that got me to this terrible place it said could not find the borg.config file. So I found this and a couple of other threads and somehow while running backup, the log states this:

2024-11-07T22:00:18.530286197Z Performing backup...
2024-11-07T22:00:18.530339666Z Starting the backup...
2024-11-07T22:00:18.998125223Z Creating archive at "/mnt/borgbackup/borg::20241107_220018-nextcloud-aio"

and hangs

before I brick anything else, and without a backup, how do I restore AIO/borg so that I can start my server up again?

Edit: NextCloud doesn't start anymore, but I did find another borg backup and deleted it, the backup is running now, will update when it's finished

**Reply by NEVARLeVrai** [reply]
i dont have my borg config anymore how to restore ???
